### PR TITLE
Add system.ip field to context

### LINKED ISSUE #349: Add agent IP address to context
It would be really useful to collect source host / IP address from agents on every request. Apart from having the info, this feature would allow `add_kubernetes_metadata` to match originating pod and enrich events with metadata from Kubernetes.

I've been playing with that idea and did this:

https://github.com/elastic/apm-server/compare/master...exekias:system-ip?expand=1

along with these settings:
```
processors:
- add_kubernetes_metadata:
    indexers:
      - ip_port:
    matchers:
      - fields:
          lookup_fields: ["context.system.ip"]
```

so far it works.

### PR DESCRIPTION
**Pull request by exekias**
This PR adds `context.system.ip` field. This expose IP of the app serving received traces and will enable matching it by source in the future

closes #349

### CODE REVIEW

**Comment by exekias** [comment]
Did a 2nd attempt, let me know what you think, as I needed to update processors to inject HTTP request, so they can extract info from it.

First try (5405257) was simpler but broke tests as it didn't follow the convention on how processors are tested.

**Comment by simitt** [comment]
@exekias all the other fields regarding [system](https://github.com/elastic/apm-server/blob/master/docs/spec/system.json) are collected directly by the agents and sent to the server. I expected the `system.ip` to also be collected by agents. Could you please elaborate a bit on why you would handle this different?

File: beater/handlers.go [hunk 1]
```diff
@@ -146,42 +145,14 @@ func ipRateLimitHandler(rateLimit int, h http.Handler) http.Handler {
 	}
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if deny(extractIP(r)) {
+		if deny(utility.ExtractIP(r)) {
```
Comment (simitt) [inline beater/handlers.go]: nice improvement!

**Comment by exekias** [comment]
closing in favor of #393

**Pull request closed without merging**